client/components/navbar/navbar-notification.service.js

```diff
@@ -21,33 +21,31 @@ class NavbarNotificationService {
   getSubLinks() {
     return this.getMessages()
       .then(messages => messages.map(message => this.convertSubLink(message)))
-      .catch(() => undefined);
+      .catch(() => this.$q.when([]));
   }
 
-  /* eslint-disable */
-  _formatTime(dateTime) {
+  static _formatTime(dateTime) {
     return moment(dateTime).fromNow();
   }
 
   _toggleSublinkAction(toUpdate, linkClicked) {
     if (toUpdate.isActive && !toUpdate.updating) {
-      toUpdate.updating = true;
+      _.set(toUpdate, 'updating', true);
       this.OvhApiNotificationAapi.post({ completed: [toUpdate.id] }).$promise.then(() => {
-        toUpdate.isActive = false;
-        toUpdate.acknowledged = true;
-      }).finally(() => { toUpdate.updating = false; });
+        _.set(toUpdate, 'isActive', false);
+        _.set(toUpdate, 'acknowledged', true);
+      }).finally(() => { _.set(toUpdate, 'updating', false); });
     } else if (!toUpdate.isActive && !toUpdate.updating && !linkClicked) {
-      toUpdate.updating = true;
+      _.set(toUpdate, 'updating', true);
       this.OvhApiNotificationAapi.post({ acknowledged: [toUpdate.id] }).$promise.then(() => {
-        toUpdate.isActive = true;
-        toUpdate.acknowledged = true;
-      }).finally(() => { toUpdate.updating = false; });
+        _.set(toUpdate, 'isActive', true);
+        _.set(toUpdate, 'acknowledged', true);
+      }).finally(() => { _.set(toUpdate, 'updating', false); });
```

Comment:
Why so many _.set?

Reply:
due to: https://eslint.org/docs/rules/no-param-reassign